File: swara-music/src/main/java/com/swara/music/writers/StringWriter.java
package com.swara.music.writers;

import java.io.OutputStream;

import com.swara.music.MusicWriter;
import com.swara.music.elements.Song;

/**
 * Writes a {@link Song} to a standard Java String. This writer is included primarily for debugging
 * and logging purposes as well as the gold-standard for performance; it doesn't get faster than
 * dumping an object to string!
 */
public final class StringWriter implements MusicWriter {

    @Override
    public void write(OutputStream out, Song song) throws Exception {
        out.write(song.toString().getBytes());
    }

}

File: swara-music/src/main/scala/com/swara/music/writers/StringWriter.scala
package com.swara.music
package writers

import java.io.OutputStream
import scala.util.Try

import com.swara.music.Song

/**
 * Writes a [[Song]] to a standard Java String. This writer is included primarily for debugging
 * and logging purposes as well as the gold-standard for performance; it doesn't get faster than
 * dumping an object to string!
 */
object StringWriter extends MusicWriter {

  override def write(song: Song, out: OutputStream): Try[Unit] =
    Try(out.write(song.toString.getBytes))

}